### DISCUSSION TITLE
[rocblas] Can't build due to out of memory

### DISCUSSION THREAD

**Discussion by petronny** [rocm-arch/rocm-arch#834] [Support] [answered]
Hi, I've tried many times to build rocblas but the [build always runs out of the memory](https://github.com/arch4edu/cactus/runs/7382253691?check_suite_focus=true) (error code 137 means OOM) on a build machine with 16GiB RAM.

Do you have any idea about how much RAM the build actually needs or any workaround?

PS. I've set `_tensile_threads=1` but it still goes OOM.

**Comment by tpkessler** [comment, marked as answer]
Hi @petronny! In my experience, the problem is not the number of threads but the single core compilation of the C++ code generated by tensile. I usually need 40 GiB (RAM plus swap) to build rocblas.

**Reply by kode54** [reply]
I can't even build rocm-llvm with 16GiB of RAM and 32GiB of swap.

**Reply by petronny** [reply]
rocm-llvm should be easier. It could be even [compiled in the default GitHub Action runner](https://github.com/arch4edu/cactus/actions/runs/3075678432).

You may try using less cores to save memory.

**Reply by kode54** [reply]
It's failing a link step, which needs all the memory for a single core.
